Social Anxiety rework

RaveRadbury opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request
- This PR moves `on_process` features of the social anxiety quirk over to a `handle_speech` proc.
- In the `handle_speech` proc is a modified version of stuttering that activates per word and also doesn't drop letters
- Adds mood and sanity modifiers to complement the use of oview in determining if a social anxiety event should happen
- Adds a chance for filler words like "um" to be added between words
- Chances to stop speaking mid-sentence or have your whole statement replaced with a single filler word
- Adds another chance to be silenced, reduces the length of silencing on both events.
- Retreating into yourself now requires a failed speech check *and* a mood check.

Neglecting mood and sanity will have a noticeable effect on your speaking even if you're in isolation. The mechanics reinforce using shorter sentences with fewer words to prevent getting cut off mid-sentence when social anxiety is exacerbated.

## Why It's Good For The Game
Social anxiety `on_process` is currently stammering and silence glued to each other with a fun little message that procs at the beginning of the round once (usually when you're alone) and also spaghetti meme joke.

Players currently find it unfair to get hit with 10 silence when they are alone or with one other person, additionally the 10 silence feels on the long side for how often it happens. The silence can't even be stopped by `FEARLESS` once it has been applied.

These changes will make social anxiety more dependent on mood, sanity, and actually being around other crew. With this new system, a social anxiety sufferer can actually walk away from a group of people and have their speaking improve, rather than being at the mercy of constantly running `on_process`


I'd like to add a forced whisper event but I'm not sure how I'd go about that, if anyone knows how to go about that let me know.

I'm also not firm on a lot of the balancing here... gonna need to work it over some more.

## Changelog
:cl:
qol: Social Anxiety is more immersive
balance: made social anxiety more tied to speaking, along with more affect from sanity, mood, and nearby people
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

RaveRadbury committed:
Reworks social anxiety

on_proccess to handle_speech, per-word stutter, filler words, chokes

RaveRadbury committed:
flavor text and silence stack tweak

Mothblocks commented on code/datums/quirks/negative.dm:
```diff
@@ -422,28 +422,68 @@
+/datum/quirk/social_anxiety/proc/handle_speech(datum/source, list/speech_args)
 	if(HAS_TRAIT(quirk_holder, TRAIT_FEARLESS))
 		return
+	var/datum/component/mood/mood = quirk_holder.GetComponent(/datum/component/mood)
```
Mood component is not guaranteed (mood is in itself a config option). You need to handle the case where it does not exist.

Also, CC @ninjanomnom on GetComponent usage.

RaveRadbury replied:
Thinking about this.

If mood component isn't guaranteed due to config I'm more or less looking at it being limited to a *modifier* to `people_nearby` and I need to balance for `people_nearby` with a possibility of mood being null.

Not sure that will really add anything? I'm thinking like a 1.5 to 0.5 possible modifier from a combination of mood and sanity.

MrMelbert replied:
Worst case, could just set Social Anxiety to be `mood_quirk = TRUE`, which means it can only be selected if mood is enabled in the config.

optimumtact replied:
GetComponent is justified, stop acting like it's the end of the world, that leads to stupid signal abuse

RaveRadbury replied:
Another option could be to use hunger as a modifier when mood is not available.

Mothblocks replied:
I'm pretty openly "GetComponent is alright", though there's undeniably plenty of cases where it is not. That's why I CC'd ninjanomnom for an audit--getting the mood component is how most other code like this works.

Mothblocks commented on code/datums/quirks/negative.dm:
```diff
@@ -422,28 +422,68 @@
+	if(message)
+		var/list/message_split = splittext(message, " ")
+		var/list/new_message = list()
+		var/mob/living/carbon/human/Q = quirk_holder
```
https://github.com/tgstation/tgstation/blob/master/.github/CONTRIBUTING.md#use-descriptive-and-obvious-names

MrMelbert commented on code/datums/quirks/negative.dm:
```diff
@@ -422,28 +422,68 @@
 	UnregisterSignal(quirk_holder, list(COMSIG_MOB_EYECONTACT, COMSIG_MOB_EXAMINATE))
 
-/datum/quirk/social_anxiety/on_process(delta_time)
+/datum/quirk/social_anxiety/proc/handle_speech(datum/source, list/speech_args)
 	if(HAS_TRAIT(quirk_holder, TRAIT_FEARLESS))
 		return
```
```suggestion
/datum/quirk/social_anxiety/proc/handle_speech(datum/source, list/speech_args)
	SIGNAL_HANDLER

	if(HAS_TRAIT(quirk_holder, TRAIT_FEARLESS))
		return
```

MrMelbert commented on code/datums/quirks/negative.dm:
```diff
@@ -422,28 +422,68 @@
+	RegisterSignal(quirk_holder, COMSIG_MOB_SAY, .proc/handle_speech)
 
 /datum/quirk/social_anxiety/remove()
 	UnregisterSignal(quirk_holder, list(COMSIG_MOB_EYECONTACT, COMSIG_MOB_EXAMINATE))
```
UnregisterSignal(quirk_holder, list(COMSIG_MOB_EYECONTACT, COMSIG_MOB_EXAMINATE))
```suggestion
	UnregisterSignal(quirk_holder, list(COMSIG_MOB_EYECONTACT, COMSIG_MOB_EXAMINATE, COMSIG_MOB_SAY))
```

RaveRadbury committed:
adds comsig_mob_say to signal unregister

thanks melbert

Co-authored-by: MrMelbert <51863163+MrMelbert@users.noreply.github.com>

RaveRadbury committed:
adds signal handler

SIGNAL_HANDLER is a macro thrown onto signal procs to make sure they don't sleep for technical reasons.

Co-authored-by: MrMelbert <51863163+MrMelbert@users.noreply.github.com>

RaveRadbury committed:
changes variable names to descriptive

RaveRadbury committed:
Merge branch 'Social-Anxiety-Retool' of https://github.com/RaveRadbury/tgstation into Social-Anxiety-Retool

RaveRadbury commented:
Gonna try and sort this today if I can, still need to hear from ninjanomnom about component usage.

RaveRadbury committed:
adds mood failsafe, retools event odds

RaveRadbury committed:
fiddling, finessing

RaveRadbury commented:
No longer stale

RaveRadbury committed:
removes duplicate line

whoopsie
